form/form.go

```diff
@@ -357,30 +378,28 @@ func makeStructEncoder(t reflect.Type) *structEncoder {
 			continue
 		}
 
+		fldTyp := reflectField.Type
+		fldKind := fldTyp.Kind()
+
 		if Strict && options != nil &&
 			(options.Empty || options.Invert || options.Zero) &&
-			reflectField.Type.Kind() != reflect.Bool {
+			fldKind != reflect.Bool {
 
 			panic(fmt.Sprintf(
 				"Cannot specify `empty`, `invert`, or `zero` for non-boolean field; on: %s/%s",
 				t.Name(), reflectField.Name,
 			))
 		}
 
-		var isPtr bool
-		if reflectField.Type.Kind() == reflect.Ptr {
-			isPtr = true
-		}
-
 		se.fields = append(se.fields, &field{
 			formName:   formName,
 			index:      i,
-			isAppender: isAppender(reflectField.Type),
-			isPtr:      isPtr,
+			isAppender: isAppender(fldTyp),
+			isPtr:      fldKind == reflect.Ptr,
```

Comment:
Nice improvement. Somehow I'd convinced myself that you weren't allowed to specify an expression like this.